main/controllers/dbController.js

```diff
@@ -0,0 +1,84 @@
+const db = require("../models/db");
+
+const dbController = {};
+
+// postgres - Connections:
+// db: clusterID, userID, clusterName, ports, created_on
+// primary key: clusterID + userID
+
+
+dbController.saveConnection = async (req, res, next) => {
+
+  try {
+    const { id, name, uri, ports, created, userID } = req.body;
+
+    const portsJSON = JSON.stringify(ports);
+    const query = 'INSERT INTO "Connections" (cluster_id, user_id, cluster_name, cluster_uri, ports, created_on) VALUES ($1, $2, $3, $4, $5, $6)';
+    const values = [id, userID, name, uri, portsJSON, created]
+
+    const response = await db.query(query, values);
+
+    res.locals.response = response;
+    return next();
+
+  }
+  catch (error) {
+    console.error(error);
+    return next(error);
+  }
+}
+
+dbController.getConnections = async (req, res, next) => {
+  try {
+
+    const userid = req.params.userid;
+
+    const query = 'SELECT * FROM "Connections" WHERE user_id=$1';
+    const value = [userid]
+
+    const response = await db.query(query, value);
+
+    const data = response.rows
+    res.locals.data = data;
+    return next();
+  }
+  catch (error) {
+    console.error(error);
+    return next(error);
+
+  }
+}
+
+dbController.deleteConnections = async (req, res, next) => {
+
+  try {
+    // req.body should contain two pieces of information: userid and the array of ports.
+    const { userid, clusters } = req.body;
+    // we'll first delete those particular rows from the database
+
+    const clustersQuery = clusters.map((node, index) => {
+      return `$${index + 2}`
```

Comment:
why + 2?

Reply:
Clusters is an array of ID numbers, but I needed them to be mapped to the query variables $2, $3, $4, etc. UserID is always going to be $1. So index 0 becomes $2, index 1 becomes $3, etc.